node/alphanet/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+extern crate env_logger;
+#[macro_use]
+extern crate log;
+extern crate serde;
+extern crate serde_derive;
+
+use std::{cmp, env, thread};
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::time::{Duration, Instant};
+
+use env_logger::Builder;
+use futures::future::Future;
+use futures::sink::Sink;
+use futures::sync::mpsc;
+use tokio::timer::Delay;
+
+use chain::ChainPayload;
+use network::nightshade_protocol::{spawn_consensus_network, start_peer};
+use nightshade::nightshade_task::{Control, spawn_nightshade_task};
+use primitives::aggregate_signature::{BlsPublicKey, BlsSecretKey, get_bls_key_pair};
+use primitives::hash::hash_struct;
+use primitives::network::PeerInfo;
+use primitives::signature::{get_key_pair, PublicKey, SecretKey};
+use primitives::types::AccountId;
+
+#[allow(dead_code)]
+fn configure_logging(log_level: log::LevelFilter) {
+    let internal_targets = vec![
+        "nightshade",
+        "alphanet",
+    ];
+    let mut builder = Builder::from_default_env();
+    internal_targets.iter().for_each(|internal_targets| {
+        builder.filter(Some(internal_targets), log_level);
+    });
+
+    let other_log_level = cmp::min(log_level, log::LevelFilter::Info);
+    builder.filter(None, other_log_level);
+
+    if let Ok(lvl) = env::var("RUST_LOG") {
+        builder.parse(&lvl);
+    }
+    if let Err(e) = builder.try_init() {
+        warn!(target: "client", "Failed to reinitialize the log level {}", e);
+    }
+}
+
+// TODO: Explain this function and their arguments
+fn run_node(authority: usize,
```

Comment:
Also, for consistency, I suggest following the existing naming, e.g. name this method `start_from_configs`.